CMakeLists.txt

```diff
@@ -163,6 +163,72 @@ option(ENABLE_PROBES_LINUX "build OVAL probes for the Linux OVAL tests" ${IS_LIN
 string(COMPARE EQUAL ${CMAKE_SYSTEM_NAME} "Solaris" IS_SOLARIS)
 option(ENABLE_PROBES_SOLARIS "build OVAL probes for the Solaris OVAL tests" ${IS_SOLARIS})
 
+if(ENABLE_PROBES)
+	if(ENABLE_PROBES_INDEPENDENT)
+		set(INDEPENDENT_ENVIRONMENTVARIABLE_PROBE TRUE)
+		set(INDEPENDENT_ENVIRONMENTVARIABLE58_PROBE TRUE)
+		set(INDEPENDENT_FAMILY_PROBE TRUE)
+		if(GCRYPT_FOUND OR NSS_FOUND)
+			set(INDEPENDENT_FILEHASH_PROBE TRUE)
+			set(INDEPENDENT_FILEHASH58_PROBE TRUE)
+		endif()
+		if(OPENDBX_FOUND)
+			set(INDEPENDENT_SQL_PROBE TRUE)
+			set(INDEPENDENT_SQL57_PROBE TRUE)
+		endif()
+		set(INDEPENDENT_SYSTEM_INFO_PROBE TRUE)
+		set(INDEPENDENT_TEXTFILECONTENT_PROBE TRUE)
+		set(INDEPENDENT_TEXTFILECONTENT54_PROBE TRUE)
+		set(INDEPENDENT_VARIABLE_PROBE TRUE)
+		set(INDEPENDENT_XMLFILECONTENT_PROBE TRUE)
+	endif()
+	if(ENABLE_PROBES_UNIX)
+		set(UNIX_DNSCACHE_PROBE TRUE)
+		set(UNIX_FILE_PROBE TRUE)
+		set(UNIX_FILEEXTENDEDATTRIBUTE_PROBE TRUE)
+		if(GCONF_FOUND)
+			set(UNIX_GCONF_PROBE TRUE)
+		endif()
+		set(UNIX_INTERFACE_PROBE TRUE)
+		set(UNIX_PASSWORD_PROBE TRUE)
+		set(UNIX_PROCESS_PROBE TRUE)
+		if(CAP_FOUND)
+			set(UNIX_PROCESS58_PROBE TRUE)
+		endif()
+		set(UNIX_ROUTINGTABLE_PROBE TRUE)
+		set(UNIX_RUNLEVEL_PROBE TRUE)
+		set(UNIX_SHADOW_PROBE TRUE)
+		set(UNIX_SYMLINK_PROBE TRUE)
+		set(UNIX_SYSCTL_PROBE TRUE)
+		set(UNIX_UNAME_PROBE TRUE)
+		set(UNIX_XINETD_PROBE TRUE)
+		if(ENABLE_PROBES_LINUX)
+			if(APTPKG_FOUND)
+				set(LINUX_DPKGINFO_PROBE TRUE)
+			endif()
+			set(LINUX_IFLISTENERS_PROBE TRUE)
+			set(LINUX_INETLISTENINGSERVERS_PROBE TRUE)
+			if(BLKID_FOUND)
+				set(LINUX_PARTITION_PROBE TRUE)
+			endif()
+			if(RPM_FOUND)
+				set(LINUX_RPMINFO_PROBE TRUE)
+				set(LINUX_RPMVERIFY_PROBE TRUE)
+				set(LINUX_RPMVERIFYFILE_PROBE TRUE)
+				set(LINUX_RPMVERIFYPACKAGE_PROBE TRUE)
+			endif()
+			if(SELINUX_FOUND)
+				set(LINUX_SELINUXBOOLEAN_PROBE TRUE)
+				set(LINUX_SELINUXSECURITYCONTEXT_PROBE TRUE)
+			endif()
+			if(DBUS_FOUND)
+				set(LINUX_SYSTEMDUNITDEPENDENCY_PROBE TRUE)
+				set(LINUX_SYSTEMDUNITPROPERTY_PROBE TRUE)
+			endif()
+		endif()
+	endif()
+endif()
```

Comment:
this approach won't work, you need to use dependent option as we discussed at the meeting. These can't be changed at configure time and the approach with the conditionals just can't work, it doesn't override to false, it overrides to true.

Reply:
These options are already dependent on found dependencies (SELinux, DBus, RPM, ...). How it won't work? What do you suggest instead?

Reply:
They are not options, in other words you can't choose to disable any of the probes. This might be fine for this PR but we should do this properly in the future.